tests/testthat/test-connect_target_timeseries.R

```diff
@@ -436,3 +436,181 @@ test_that("connect_target_timeseries on HIVE-PARTTIONED timeseries data works on
+        observation = "double"
+      )
+    )
+  }
+)
+
+test_that(
+  "connect_target_timeseries works with multi-file SubTreeFileSystem hub",
+  {
+    skip_if_offline()
+    fs::dir_delete(ts_dir_hub_path)
+    fs::dir_copy(ts_hub_path, ts_dir_hub_path)
+    ts_path <- validate_target_data_path(ts_dir_hub_path, "time-series")
+    # Read timeseries data from single file
+    ts_dat <- arrow::read_csv_arrow(ts_path)
+    # Delete single time-series file in preparation for creating time-series directory
+    fs::file_delete(ts_path)
+
+    # Create a seperate file for each target in a time-series directory
+    ts_dir <- fs::path(ts_dir_hub_path, "target-data", "time-series")
+    fs::dir_create(ts_dir)
+    split(ts_dat, ts_dat$target) |> purrr::iwalk(
+      ~ {
+        target <- gsub(" ", "_", .y)
+        path <- file.path(ts_dir, paste0("target-", target, ".csv"))
+        arrow::write_csv_arrow(.x, file = path)
+      }
+    )
+
+    hub_path <- withr::local_tempdir()
+    # Create a SubTreeFileSystem hub to mimick cloud hub and copy example hub contents
+    # into it
+    loc_fs <- arrow::SubTreeFileSystem$create(hub_path)
+    arrow::copy_files(ts_dir_hub_path, loc_fs)
+    config_tasks <- read_config(hub_path)
+
+    # mock read_config function as it assumes any SubTreeFileSystem hub is an s3 bucket
+    # and creates `s3` prefixed paths to URIs.
+    local_mocked_bindings(
+      read_config = function(...) {
+        config_tasks
+      }
+    )
+
+    # TESTS ====
+    # Connect to time-series data
+    ts_con <- connect_target_timeseries(loc_fs)
+    expect_s3_class(ts_con,
+      c(
+        "target_timeseries", "FileSystemDataset", "Dataset", "ArrowObject",
+        "R6"
+      ),
+      exact = TRUE
+    )
+
+    # Check files opened correctly as ts_path captured correctly
+    expect_equal(
+      basename(attr(ts_con, "ts_path")),
+      "time-series"
+    )
+    expect_length(ts_con$files, 2L)
+    expect_equal(
+      basename(ts_con$files),
+      basename(fs::dir_ls(ts_dir, recurse = TRUE, type = "file"))
+    )
+
+    expect_equal(
+      ts_con$schema$ToString(),
+      "date: date32[day]\ntarget: string\nlocation: string\nobservation: double"
+    )
+
+    # Test the collect method
+    all <- dplyr::collect(ts_con)
+
+    expect_equal(dim(all), c(20510L, 4L))
+    expect_s3_class(all, "tbl_df", exact = FALSE)
+    expect_equal(names(all), c("date", "target", "location", "observation"))
+    expect_equal(
+      unique(all$location),
+      c(
+        "01", "15", "18", "27", "30", "37", "48", "US", "32", "20",
+        "17", "29", "41", "04", "06", "13", "19", "21", "22", "24", "23",
+        "26", "28", "38", "31", "34", "39", "40", "42", "72", "45", "51",
+        "53", "55", "54", "56", "44", "05", "12", "16", "35", "36", "47",
+        "02", "09", "50", "08", "11", "10", "25", "33", "46", "49"
+      )
+    )
+    expect_equal(
+      unique(all$target), c("wk flu hosp rate", "wk inc flu hosp")
+    )
```

Comment:
avoids a situation where the upstream data are re-ordered

```suggestion
    expect_setequal(
      unique(all$location),
      c(
        "01", "15", "18", "27", "30", "37", "48", "US", "32", "20",
        "17", "29", "41", "04", "06", "13", "19", "21", "22", "24", "23",
        "26", "28", "38", "31", "34", "39", "40", "42", "72", "45", "51",
        "53", "55", "54", "56", "44", "05", "12", "16", "35", "36", "47",
        "02", "09", "50", "08", "11", "10", "25", "33", "46", "49"
      )
    )
    expect_setequal(
      unique(all$target), c("wk flu hosp rate", "wk inc flu hosp")
    )
```